compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/repositories/FileSystemRepository.java

```diff
@@ -172,9 +173,54 @@ public FileSystemRepository(Environment environment, Path cacheDirectory, String
         } catch (IOException e) {
             throw new RuntimeException("Error while accessing Distribution cache: " + e.getMessage());
         }
+
+        List<Path> incompatibleVersions = new ArrayList<>();
+
+        if (!versions.isEmpty()) {
+            for (Path ver : versions) {
+                Path pkgJsonPath = bala.resolve(org.value()).resolve(name.value()).resolve(ver.toString())
+                        .resolve("any").resolve(ProjectConstants.PACKAGE_JSON);
```

Comment:
The platform can be `any` or `java11`. Shall we try to make use of https://github.com/ballerina-platform/ballerina-lang/blob/5254c78fc787190bcd9c36135368ecc900803e6a/compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/repositories/FileSystemRepository.java#L242 method to get the bala path?

Reply:
Made use of `getPackagePath` to resolve platform